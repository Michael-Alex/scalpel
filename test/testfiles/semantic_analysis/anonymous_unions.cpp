class c
{
	union
	{
		double d;
		int i;
	} ;
};

static union
{
	double d;
	int i;
};

