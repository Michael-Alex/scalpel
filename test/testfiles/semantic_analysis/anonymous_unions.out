{
  "id": "namespace-0",
  "classes":
  [
    {
      "id": "class-0",
      "name": "c",
      "anonymous unions":
      [
        {
          "id": "anonymous-member-union-0",
          "access": "private",
          "variables":
          [
            {
              "id": "member-variable-0",
              "name": "d",
              "access": "public",
              "type":
              {
                "fundamental type": "double"
              }
            },
            {
              "id": "member-variable-1",
              "name": "i",
              "access": "public",
              "type":
              {
                "fundamental type": "int"
              }
            }
          ]
        }
      ],
      "destructor":
      {
        "access": "public"
      },
      "variables":
      [
        {
          "id": "member-variable-2",
          "access": "private",
          "type":
          {
            "anonymous member union": "anonymous-member-union-0"
          }
        }
      ]
    }
  ],
  "anonymous unions":
  [
    {
      "id": "anonymous-union-0",
      "variables":
      [
        {
          "id": "member-variable-3",
          "name": "d",
          "access": "public",
          "type":
          {
            "fundamental type": "double"
          }
        },
        {
          "id": "member-variable-4",
          "name": "i",
          "access": "public",
          "type":
          {
            "fundamental type": "int"
          }
        }
      ]
    }
  ],
  "variables":
  [
    {
      "id": "variable-0",
      "static": "true",
      "type":
      {
        "anonymous union": "anonymous-union-0"
      }
    }
  ]
}
